web_monitoring/diffing_server.py

```diff
@@ -124,6 +136,23 @@ def write_error(self, status_code, **kwargs):
             self.set_status(response['code'])
         self.finish(response)
 
+    def check_response_for_error(self, response):
+        #Check if the HTTP requests were successfull and handle exeptions
+        if response.error is not None:
+            try:
+                response.rethrow()
+            except (ValueError, IOError):
+                #Response code == 599 means that no HTTP response was received.
+                #In this case the error code should become 400 indicating that the error was
+                #raised because of a bad request parameter.
```

Comment:
There is actually a specific HTTP error for this case: `504 GATEWAY TIMEOUT` https://httpstatuses.com/504

Reply:
If no HTTP response is received tornado seems to return 599, according to the docs. http://www.tornadoweb.org/en/stable/httpclient.html#tornado.httpclient.HTTPError
How should we go about this? :)

Reply:
Right, Tornado’s doing the correct thing here. Sorry if I was unclear — I didn’t mean we shouldn’t be checking for a 599; I meant our server should return a 504 when it receives a 599. 504 specifically means “this server timed out when asking another server for the data you requested.” It’s a more specific code than 400 in this case.

That said, it would likely be good to make sure here that we really are dealing with a timeout and not something else, like a malformed address that was never attempted (that should probably still be a 400 or maybe a 422). For example, I *think* your `test_invalid_url_a_format` test is triggering a `socket.gaierror` here (subclass of `OSError`), which would be a 400/422 because it’s a bad address, not a timeout.

Reply:
Either way, I’m not too worried about sending exactly the right code here; just handling this case better than we previously were is probably an improvement. We can do a separate PR later for more specific and correct codes here.